sdk/go/common/resource/plugin/provider.go

```diff
@@ -151,6 +151,28 @@ func (d DiffKind) IsReplace() bool {
 	}
 }
 
+// ToReplace converts a DiffKind into the equivalent replacement if it not already
+// a replacement.
+func (d DiffKind) ToReplace() DiffKind {
```

Comment:
Nit: maybe `AsReplace`? `ToReplace` sounds like a question to me for some reason. Totally minor, though--feel free to leave things as-is if you'd like.